torch/_inductor/compile_fx_async.py

```diff
@@ -0,0 +1,169 @@
+from __future__ import annotations
+
+from collections import namedtuple
+from typing import Any, Callable, Optional, TYPE_CHECKING
+from typing_extensions import final, override
+
+import torch._inductor.async_compile  # noqa: F401 required to warm up AsyncCompile pools
+from torch._inductor.output_code import CompiledFxGraphConstants, OutputCode
+
+from .compile_fx import _CompileFxKwargs, _InProcessFxCompile, FxCompile
+from .output_code import complex_memory_overlap as complex_memory_overlap  # noqa: F401
+
+
+if TYPE_CHECKING:
+    from collections.abc import Sequence
+    from concurrent.futures import Future
+
+    from torch._inductor.utils import BoxedBool, InputType
+    from torch.fx import GraphModule
+
+    from .compile_fx_ext import _OutOfProcessFxCompile, _WireProtocolPickledOutput
+
+
+_PostCompileData = namedtuple(
+    "_PostCompileData", ("example_inputs", "cudagraphs", "constants")
+)
+
+
+# _AsyncOutputCode handles the actual management of waiting for an
+# out-of-process compile to finish and then switching over to it.
+@final
+class _AsyncOutputCode(OutputCode):
+    _eager_forward: Optional[Callable[..., Any]]
+    _output_code: Optional[OutputCode]
+    _future: Optional[Future[_WireProtocolPickledOutput]]
+    _callback: Callable[[_WireProtocolPickledOutput], OutputCode]
+    _post_compile_data: Optional[_PostCompileData] = None
+
+    def __init__(
+        self,
+        # eager_forward is run until the future is finished.
+        eager_forward: Callable[..., Any],
+        # this responds with the result of the out-of-process compile when it's
+        # ready.
+        future: Future[_WireProtocolPickledOutput],
+        # this callback gets called to turn the _WireProtocolPickledOutput into an OutputCode
+        callback: Callable[[_WireProtocolPickledOutput], OutputCode],
+    ) -> None:
+        self._eager_forward = eager_forward
+        self._output_code = None
+
+        self._future = future
+        self._callback = callback
+
+        # This tells callers to call us with `inputs` instead of `*inputs`.
+        self._boxed_call = True
```

Comment:
I know that all inductor generated artifacts will always return with _boxed_call = True today, but shouldn't we check whether the produced compiled_artifact has _boxed_call = True, and thread that value through here?